controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -126,6 +157,16 @@ public void createStream(final String scopeName, final CreateStreamRequest creat
             LoggerHelpers.traceLeave(log, "createStream", traceId);
             return;
         }
+
+        try {
+            authenticate(scopeName + "/" + createStreamRequest.getStreamName(), READ_UPDATE);
```

Comment:
for create stream, shouldn't the `READ_UPDATE` be required on the scope, instead of the scope/stream   (the stream doesn't exist yet).

Reply:
This is too wide privileges. Either the admin needs to open the whole scope OR only few users can create streams. Because of this the decision was taken that we can add stream names.
Using this approach both cases will be covered. People who have write access to scope can create the stream. Admin can give access to users to create and use just one stream too.

Reply:
Understood.